Assets/Scripts/CameraHandler.cs

```diff
@@ -16,11 +16,26 @@ private void Awake()
         _virtualCamera = GetComponent<CinemachineVirtualCamera>();
     }
 
+    private void Update()
+    {
+        CheckTargetExist();
+    }
+
+    private void CheckTargetExist()
+    {
+        if (_virtualCamera.Follow == null)
+        {
+            FindNewTarget();
+        }
```

Comment:
instead of this, you could have this in update and name the `_virtualCamera.Follow == null` `HasPlayerToFollow() == false`

Reply:
Remove camera generally so don't check it